components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -393,8 +414,14 @@ public Factory recordStats(boolean recordStats) {
             return this;
         }
 
+        public Factory connectionDecorator(ConnectionDecorator connectionDecorator) {
+            this.connectionDecorator = connectionDecorator;
+            return this;
+        }
+
         public SimpleConnectionPool create(Origin origin) {
-            return new SimpleConnectionPool(origin, connectionPoolSettings, connectionFactory, recordStats);
+            return new SimpleConnectionPool(origin, connectionPoolSettings, connectionFactory, recordStats,
+                    connectionDecorator);
         }
     }
 }
```

Comment:
Now after the latest round of changes, it is apparent that the `SimpleConnectionPool` class doesn't need to change at all. The connections can be decorated by the `ConnectionFactory`. Please consider implementing a new connection factory that delegates connection creation into `NettyConnectionFactory` and decorates the connections with connection expiry behaviour.